core/src/main/java/org/bitcoinj/core/TransactionInput.java

```diff
@@ -131,28 +131,24 @@ public TransactionInput(@Nullable Transaction parentTransaction, byte[] scriptBy
     private TransactionInput(@Nullable Transaction parentTransaction, byte[] scriptBytes,
                             TransactionOutPoint outpoint, long sequence, @Nullable Coin value) {
         checkArgument(value == null || value.signum() >= 0, () -> "value out of range: " + value);
+        parent = parentTransaction;
         this.scriptBytes = scriptBytes;
         this.outpoint = outpoint;
         this.sequence = sequence;
         this.value = value;
-        setParent(parentTransaction);
```

Comment:
Not sure about this one. As long as `setParent` exists, we should probably use it. We could however try to get rid of `setParent` entirely, and ideally `parent` as well at some point.